### ISSUE TITLE
bug in `stumpy/snippet.py`

### ISSUE THREAD

**Issue by NimaSarajpoor** [TDAmeritrade/stumpy#677]
I think there is a ~minor~ bug in `stumpy/snippet.py`. one can see the bug by setting param `s` to a value less than or equal to `m / 2`.

To reproduce the error:
```
T = np.random.rand(20)
m = 5
k = 2
s=2
stumpy.snippets(T, m, k=k, s=s)
```

**Comment by seanlaw** [comment]
[This](https://github.com/TDAmeritrade/stumpy/pull/878#issuecomment-1632235605) may possibly be related

**Comment by NimaSarajpoor** [comment]
Note:
I think the bug mentioned in the first post, i.e.

> I think there is a minor bug in stumpy/snippet.py. one can see the bug by setting param s to a value less than or equal to m / 2.

is now resolved: see #823 in which the slicing (of arrays) is changed from `start : start + s` to `start : stop - s + 1` (see the changes in the snippet module)

Having said that, a recent error is observed:
 
> [This](https://github.com/TDAmeritrade/stumpy/pull/878#issuecomment-1632235605) may possibly be related:
> 
> https://github.com/TDAmeritrade/stumpy/actions/runs/5520192550/jobs/10066477926

I will check that out.

**Comment by NimaSarajpoor** [comment]
> Having said that, a recent error is observed:
> 
> > [This](https://github.com/TDAmeritrade/stumpy/pull/878#issuecomment-1632235605) may possibly be related:
> > https://github.com/TDAmeritrade/stumpy/actions/runs/5520192550/jobs/10066477926
> 
> I will check that out.

[update] 
As discussed in #881, the issue in #881 is neither `self-match` nor `s < m/2`. Hence, I think it should be safe to close this issue as it has been addressed already.
